### PR TITLE
Fix some buildbot tests under Python 3

### PR DESCRIPTION
**Pull request by rodrigc**
These tests now work under Python 3:

```
trial buildbot.test.unit.test_steps_transfer
trial buildbot.test.unit.test_scripts_tryserver
trial buildbot.test.unit.test_schedulers_trysched
```

### CODE REVIEW

File: master/buildbot/process/buildstep.py [hunk 1]
```diff
@@ -812,7 +813,8 @@ def addLogWithFailure(self, why, logprefix=""):
         # helper for showing exceptions to the users
         try:
             yield self.addCompleteLog(logprefix + "err.text", why.getTraceback())
-            yield self.addHTMLLog(logprefix + "err.html", formatFailure(why))
+            yield self.addHTMLLog(logprefix + "err.html",
+                                  bytes2NativeString(formatFailure(why)))
```
Comment (tardyp) [inline master/buildbot/process/buildstep.py]: I would expect addHTMLLog to do the ascii2unicode by itself (will probably save you a lot of fixes)

Comment (rodrigc) [inline master/buildbot/process/buildstep.py]: OK I made the change.

**Comment by rodrigc** [comment]
@tardyp Can https://github.com/buildbot/buildbot/pull/2610 and https://github.com/buildbot/buildbot/pull/2613 go in?  I've incorporated your feedback.
Future patches I have are interdependent on these patches.

Thanks.

**Comment by tardyp** [comment]
sorry, I've been struggling with the CI instabilities. need to make sure everything is okay

**Pull request merged**